libconnman-qt/networkmanager.cpp

```diff
@@ -679,25 +676,14 @@ void NetworkManager::updateServices(const ConnmanObjectList &changed, const QLis
         if (service) {
             // We don't want to emit signals at this point. Those will
             // be emitted later, after internal state is fully updated
-            disconnect(service, SIGNAL(connectedChanged(bool)),
-                this, SLOT(updateDefaultRoute()));
-            disconnect(service, SIGNAL(connectedChanged(bool)),
-                m_priv, SLOT(onWifiConnectedChanged()));
-            disconnect(service, SIGNAL(connectingChanged()),
-                m_priv, SLOT(onWifiConnectingChanged()));
-            disconnect(service, SIGNAL(connectedChanged(bool)),
-                m_priv, SLOT(onEthernetConnectedChanged()));
+            disconnect(service, nullptr, m_priv, nullptr);
             service->updateProperties(obj.properties);
         } else {
             service = new NetworkService(path, obj.properties, this);
             m_servicesCache.insert(path, service);
             addedServices.append(path);
         }
 
-        // Re-connect updateDefaultRoute slot
-        connect(service, SIGNAL(connectedChanged(bool)),
-            this, SLOT(updateDefaultRoute()));
```

Comment:
Is this correct? I cannot see any other case where this updateDefaultRoute SLOT is connected anymore after this change.

Reply:
It's called as first thing from Private::onConnectedChanged().

Reply:
So the SLOT in networkmanager.h for updateDefaultRoute became obsolete?

Reply:
Sure, can remove that.